packages/gatsby-image/src/index.js

```diff
@@ -85,6 +85,12 @@ const hasNativeLazyLoadSupport =
 const isBrowser = typeof window !== `undefined`
 const hasIOSupport = isBrowser && window.IntersectionObserver
 
+// Save-Data support: https://developers.google.com/web/fundamentals/performance/optimizing-content-efficiency/save-data/#detecting_the_save-data_setting
+const hasSaveDataEnabled = () =>
+  isBrowser &&
+  `connection` in navigator &&
+  navigator.connection.saveData === true
```

Comment:
Does this need to be a function? Can we just set `hasSaveDataEnabled` variable once and not "recalculate" it?

Reply:
I need `hasSaveDataEnabled` to be a function in order to unit test this behaviour.
How would you test it if this value is computed once and for all?

Reply:
Yeah, this would make unit testing more difficult and there is not enough upside for doing that - so let's keep it as is